cfme/generic_objects/definition/ui.py

```diff
@@ -0,0 +1,477 @@
+    """Add button group for generic object definition
+    """
+    view = navigate_to(self, 'AddButtonGroup')
+    view.fill({
+        'image': image,
+        'name': name,
+        'description': description,
+        'display': display,
+    })
+    if cancel:
+        view.cancel.click()
+    else:
+        view.add.click()
+    view = self.create_view(GenericObjectDefinitionDetailsView)
+    assert view.is_displayed
+    view.flash.assert_no_error()
+
+
+@MiqImplementationContext.external_for(GenericObjectDefinition.exists.getter, ViaUI)
+def exists(self):
+    try:
+        navigate_to(self, 'Details')
+        return True
+    except (CandidateNotFound, ItemNotFound):
+        return False
+
+
+@MiqImplementationContext.external_for(GenericObjectDefinition.generic_objects.getter, ViaUI)
+def generic_objects(self):
+    return self.collections.generic_objects
+
+
+@navigator.register(GenericObjectDefinitionCollection)
+class All(CFMENavigateStep):
+    VIEW = GenericObjectDefinitionAllView
+
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        self.prerequisite_view.navigation.select('Automation', 'Automate', 'Generic Objects')
+
+
+@navigator.register(GenericObjectDefinitionCollection)
+class Add(CFMENavigateStep):
+    VIEW = GenericObjectDefinitionAddView
+
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.toolbar.configuration.item_select('Add a new Generic Object Class')
+
+
+@navigator.register(GenericObjectDefinition)
+class Details(CFMENavigateStep):
+    VIEW = GenericObjectDefinitionDetailsView
+
+    prerequisite = NavigateToAttribute('parent', 'All')
+
+    def step(self):
+        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).click()
+
+
+@navigator.register(GenericObjectDefinition)
+class Edit(CFMENavigateStep):
+    VIEW = GenericObjectDefinitionEditView
+
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Edit this Generic Object Class')
+
+
+@navigator.register(GenericObjectDefinition)
+class Instances(CFMENavigateStep):
+    VIEW = GenericObjectInstanceAllView
+
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        self.prerequisite_view.summary('Relationships').click_at('Instances')
+
+
+@navigator.register(GenericObjectDefinition)
+class AddButtonGroup(CFMENavigateStep):
+    VIEW = GenericObjectButtonGroupAddView
+
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Add a new Button Group')
+
+
+@navigator.register(GenericObjectDefinition)
+class ButtonGroupDetails(CFMENavigateStep):
+    VIEW = GenericObjectButtonGroupDetailsView
+
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self, **kwargs):
+        if kwargs:
```

Comment:
See comments above about navigation with kwargs, and avoiding it in general.

Assuming a functional implementation is needed here, the case where there is no button group name passed during navigation needs to be handled cleanly, raising an exception when none is passed.